### ISSUE TITLE
Very high response time causing request timeout on some monitors

### ISSUE THREAD

**Issue by Jayspek** [louislam/uptime-kuma#4432]
### ⚠️ Please verify that this question has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### 📝 Describe your problem

All of my monitors usually get a response time of between 100 - 200 ms, however yesterday in the middle of the night they all started getting 7,000 - 8,000 ms.

Two of them (a TCP port monitor pointed at an ssh server and Gamedig monitor pointed at a Minecraft server) are now getting request timeout errors and being marked as down (which they are not).

I saw this issue from a while back https://github.com/louislam/uptime-kuma/issues/275 that looks similar, where the solution was a DNS caching feature. I don't see the new DNS caching setting that was apparently added, only the depreciated HTTP(s) specific one.

I didn't make any changes to the server or my how network near the time the response times increased, and I'm not seeing these latencies in anything outside uptime kuma.

Basic Ping type monitors don't seem to experience the increased latency, but everythig else I'm using (HTTP(s), Gamedig and TCP Port) does. When I create new TCP Port monitors (just for testing), the same thing happens for them as well.

### 📝 Error Message(s) or Log

```log
[MONITOR] WARN: Monitor #2 'Backup Server SSH': Pending: Request timeout
```

### 🐻 Uptime-Kuma Version

1.23.11

### 💻 Operating System and Arch

Windows

### 🌐 Browser

Chrome

### 🖥️ Deployment Environment

- Runtime: nodejs 20.11.0
- Database: Not sure? Just installed it using the provided instructions so whatever the default is
- Filesystem used to store the database on: Windows NTFS on SSD
- number of monitors: 18

**Comment by chakflying** [comment]
If the monitors that are having trouble uses a hostname instead of an IP address, then it's likely you are experience issue with the DNS server currently used.

**Comment by Jayspek** [comment]
I am using hostnames, interesting. Is there some way I could test that theory?

**Comment by CommanderStorm** [comment]
Try looking up said hostname against the DNS server of your choice.
I.e. `dig dnsserver hostname type`replacing dnsserver and hostname with the appropriate values and type being A or AAAA. (The latter to check for ipv4/6 mishaps).

If this does work try reproducing the issues on the network path or by singling out pieces of infrastructure affected

**Comment by Jayspek** [comment]
It appears dig/BIND isn't free for windows, so I used this PowerShell command instead:

(Measure-Command {Resolve-DnsName -Name mc.jaspermichal.ca -Server 1.1.1.1 -DnsOnly -NoHostsFile -Type A_AAAA}).TotalMilliseconds

Which reported approximately 80ms. Not great, but a far cry from 8,000. 
I did this test from the machine uptime Kuma is running on, so as far as I can tell uptime Kuma is the only thing experiencing the issue.

I'm unfortunately not the most knowledgeable about networking so don't know what you mean by "reproducing the issue in the network path", if so far been unable to reproduce the issue outside of uptime Kuma. Is there something else I should try?

**Comment by CommanderStorm** [comment]
So the dns adress you provided does not have an IPv6 record set (AAAA-Record) => that might be a possibility.

What is the monitor/port you are trying to monitor concretely?
Is the SSH port publicly routable?
What version of minecraft are you running in gamedig (java `minecraft`, `minecraftping` vs bedrock `mbe`,`minecraftpe`,`minecraftbe`)
Could you try `npx gamedig --type minecraft mc.jaspermichal.ca:11234` on the exact machine you are running uptime kuma on?

I cannot reproduce this latency from germany: 
![image](https://github.com/louislam/uptime-kuma/assets/26258709/39a59fe4-ad2e-4c37-9ab8-d0d879b6c450)


Sidenote:
The website behind `mc.jaspermichal.ca` loads pretty slowly
![image](https://github.com/louislam/uptime-kuma/assets/26258709/2f792ed4-4cf6-4ed9-b68c-a2e679d3b233)

**Comment by Jayspek** [comment]
The npx command reports a ping of 4, but it takes a long time to run (on the order of 20 seconds). It's a Minecraft java server, I'm using the Minecraft (2009) option in uptime Kuma.
The ssh server I'm monitoring is backup.jaspermichal.ca at port 995, it is publicly routable.
It doesn't surprise me that the website at mc.jaspermichal.ca loads slowly, it just redirects ports 80 and 443 to somewhere else.

**Comment by Jayspek** [comment]
Update, I have Windows set to use encrypted DNS. When I turn that off, the latency goes away (uptime kuma response time goes back to normal and the npx command runs quickly). I'm going to try the npx command on a different windows computer on my network to see if its the same there.

**Comment by Jayspek** [comment]
I've found the most reliable way to test this in Windows is the Test-NetConnection PowerShell command, which doesn't directly report a time but it's easy to eyeball (it runs in a couple of seconds when everything is working normally, but takes about 15-20 seconds when the problem is present).
Here is what I've found.

On my home network, any Windows machines (I don't have anything linux to test on currently) using encrypted DNS experience the increased latency. This applies to all all hostnames, not just mine. As soon as I turn encrypted DNS off, everything works fine. I'm using Cloudflare for my DNS, but its the same when I tested it with Google's DNS servers as well.
If I take a computer from my house and connected it to a different network (tested at a friend's house), encrypted DNS works fine.
So its seems it wasn't actually a problem with uptime kuma, but something to do with my home network. I'll have a look at the router and see if I can figure out whats up.

Thanks everyone for the input!
